scripts/cruisekube_diagnostics.py

```
@@ -0,0 +1,631 @@
+        nums.append(0)
+    return nums[0], nums[1], nums[2]
+
+
+def at_least(value: str, minimum: str) -> bool:
+    return parse_version(value) >= parse_version(minimum)
+
+
+def parse_duration_seconds(value: str) -> int:
+    matches = re.findall(r"(\d+)\s*([smhd])", value.strip().lower())
+    if not matches:
+        raise ValueError(f"invalid duration: {value!r}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)